charms/kserve-controller/tests/integration/test_charm.py

```diff
@@ -166,11 +186,18 @@ def assert_inf_svc_state():
     create_inf_svc()
     assert_inf_svc_state()
 
+    # Remove the InferenceService deployed in RawDeployment mode
+    lightkube_client.delete(
+        inference_service_resource, name=inf_svc_name, namespace=rawdeployment_mode_namespace
+    )
 
-#    # Remove the InferenceService deployed in RawDeployment mode
-#    lightkube_client.delete(
-#        inference_service_resource, name=inf_svc_name, namespace=rawdeployment_mode_namespace
-#    )
+    assert_deleted(
+        logger,
+        lightkube_client,
+        inference_service_resource,
+        inf_svc_name,
+        rawdeployment_mode_namespace,
+    )
```

Comment:
If we want to check this, we would have to do it in the `cleanup_namespaces_after_execution` fixture.

Reply:
We would need to modify that fixture to pass inference service resource and name. I.e. this will become a bit different fixture then.

Reply:
I don't see a problem with that, though. 
Another thing you could do is to assert that the namespace is deleted, though to be honest, having the namespace or not present in the cluster should not affect the test execution.

Reply:
Moved to fixture to ensure namespace is deleted.

Reply:
When one namespace is created there is a pod scheduled there. When we issue a delete for the namespace, it takes long time to actually for it to be deleted in GH runners. If namespace is deleted and we do not assert that it is deleted, next test will fail, because pod for the next test cannot be scheduled due to resource limitation in GH runners.